cmd.go

```diff
@@ -0,0 +1,85 @@
+package wish
+
+import (
+	"fmt"
+	"io"
+	"os/exec"
+	"runtime"
+	"time"
+
+	"github.com/charmbracelet/ssh"
+)
+
+// Command sets stdin, stdout, and stderr to the current session's PTY slave.
+//
+// If the current session does not have a PTY, it sets them to the session
+// itself.
+func Command(s ssh.Session, name string, args ...string) *Cmd {
+	c := exec.Command(name, args...)
+	pty, _, ok := s.Pty()
+	if !ok {
+		c.Stdin, c.Stdout, c.Stderr = s, s, s
+		return &Cmd{cmd: c}
+	}
+
+	return &Cmd{c, &pty}
+}
+
+// Cmd wraps a *exec.Cmd and a ssh.Pty so a command can be properly run.
+type Cmd struct {
+	cmd *exec.Cmd
+	pty *ssh.Pty
+}
+
+// SetDir set the underlying exec.Cmd env.
+func (c *Cmd) SetEnv(env []string) {
+	c.cmd.Env = env
+}
+
+// Environ returns the underlying exec.Cmd environment.
+func (c *Cmd) Environ() []string {
+	return c.cmd.Environ()
+}
+
+// SetDir set the underlying exec.Cmd dir.
+func (c *Cmd) SetDir(dir string) {
+	c.cmd.Dir = dir
+}
+
+// SetStderr conforms with tea.ExecCommand.
+func (*Cmd) SetStderr(io.Writer) {}
+
+// SetStdin conforms with tea.ExecCommand.
+func (*Cmd) SetStdin(io.Reader) {}
+
+// SetStdout conforms with tea.ExecCommand.
+func (*Cmd) SetStdout(io.Writer) {}
+
+// Run runs the program and waits for it to finish.
+func (c *Cmd) Run() error {
+	if c.pty == nil {
+		return c.cmd.Run()
+	}
+
+	if err := c.pty.Start(c.cmd); err != nil {
```

Comment:
Ah, this makes a lot of sense. Had I manually added these SysProcAttr IN ADDITION to manually setting the Std{in,out,err} equal to the PTY Slave, the bash startup would stop complaining on startup about disabling job control.

```go
        c := exec.Command("bash", "-im")
	c.Stdin, c.Stdout, c.Stderr = i.Slave, i.Slave, i.Slave
	if c.SysProcAttr == nil {
		c.SysProcAttr = &syscall.SysProcAttr{}
	}
	c.SysProcAttr.Setctty = true
	c.SysProcAttr.Setsid = true
```

https://github.com/charmbracelet/ssh/blob/8ca513fa538f/pty.go#L69
https://github.com/charmbracelet/ssh/blob/8ca513fa538f49c49f851b88a1e19a5141b52583/pty_unix.go#L68